src/Simulator/IncrementalRedstoneSimulator/IncrementalRedstoneSimulator.cpp

```diff
@@ -193,3 +192,63 @@ void cIncrementalRedstoneSimulator::Simulate(float a_dt)
 		}
 	}
 }
+
+
+
+
+
+void cIncrementalRedstoneSimulator::AddBlock(Vector3i a_Block, cChunk * a_Chunk)
+{
+	// Can't inspect block, so queue update anyway
+	if (a_Chunk == nullptr)
+	{
+		m_Data.WakeUp(a_Block);
+		return;
+	}
+
+	const auto RelPos = cChunkDef::AbsoluteToRelative(a_Block, a_Chunk->GetPos());
+	const auto CurBlock = a_Chunk->GetBlock(RelPos);
+
+	// Always update mechanisms
+	if (IsMechanism(CurBlock))
```

Comment:
Perhaps IsRedstone is better? E.g. it seems like pressure plates will never get updated as they aren't mechanisms.